### PR TITLE
[BD-6] OEP-18 and Python 3.8 tests

### PR DESCRIPTION
**Pull request by ericfab179**
Update setup.py install_requirements definition.
Update travis file using new requirements.
Update tox file using new requirements.
Create requirements folder.
create pip_tools, base, test, quality, travis and tox pip-tools requirements files.
Add makefile upgrade command.
Include requirements files generated using upgrade command.
Add openedx.yaml and include OEPs list.
Add venv to gitignore.

**Reviewers:**
- [x] @morenol
- [ ] @awais786

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @ericfab179! I've created [OSPR-4483](https://openedx.atlassian.net/browse/OSPR-4483) to keep track of it in JIRA. JIRA is a place for product owners to prioritize feature reviews by the engineering development teams.

Feel free to add as much of the following information to the ticket:
- supporting documentation
- edx-code email threads
- timeline information ("this must be merged by XX date", and why that is)
- partner information ("this is a course on edx.org")
- any other information that can help Product understand the context for the PR

All technical communication about the code itself will still be done via the GitHub pull request interface. As a reminder, [our process documentation is here](http://edx-developer-guide.readthedocs.org/en/latest/process/overview.html).

File: tox.ini [hunk 1]
```diff
@@ -1,24 +1,20 @@
...
-    coverage
+    -r{toxinidir}/requirements/test.txt
 commands =
     coverage run setup.py test
     coverage report
 
-[testenv:pycodestyle]
+[testenv:quality]
 passenv = TRAVIS TRAVIS_JOB_ID TRAVIS_BRANCH
+whitelist_externals =
```
Comment (morenol) [inline tox.ini:14]: I think that this is not needed here

**Comment by ericfab179** [comment]
@awais786 We think this PR is ready to be merged.

**Comment by awais786** [comment]
@jmbowman I don't have rights to merge this.

**Comment by jmbowman** [comment]
@awais786 I just updated the permissions, you should be able to now.

**Comment by edx-webhook** [comment]
@ericfab179 🎉 Your pull request was merged!

Please take a moment to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=edx/openedx-chem&entry.867055334=https://github.com/edx/openedx-chem/pull/2&entry.1484655318=https://github.com/ericfab179&entry.752974735=2020-05-07+00:25&entry.1917517419=2020-05-11+16:12&entry.2133058324=Yes) so we can improve your experience in the future.

**Pull request merged**